kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js

```diff
@@ -640,73 +677,92 @@ define([
-            if (jobUpdateData.job_id !== jobState.job_id) {
-                return;
-            }
+                const jobIx = `${jobState.created || 0}__${jobState.job_id}`;
+                this.jobsByRetryParent[rowId].jobs[jobIx] = jobState;
 
-            // select the appropriate row
-            try {
+                // make sure that this job is the most recent version
+                const sortedJobs = Object.keys(this.jobsByRetryParent[rowId].jobs).sort();
+                const lastJob = sortedJobs[sortedJobs.length - 1];
+                const jobUpdateData = this.jobsByRetryParent[rowId].jobs[lastJob];
+                // irrelevant update -- data is not for the most recent retry
+                if (jobUpdateData.job_id !== jobState.job_id) {
+                    return;
+                }
                 // update the row
-                this.dataTable.DataTable().row(`#job_${rowIx}`).data(jobState);
-            } catch (e) {
-                console.error('Error trying to update jobs table:', e);
-            }
+                this.dataTable.DataTable().row(`#job_${rowId}`).data(jobState);
+            });
         }
 
         /**
-         * parse and update the row with job info
+         * update the table on receiving job retry data
          * @param {object} message
          */
-        handleJobInfo(message) {
-            Object.values(message).forEach((infoMessage) => {
-                this._handleJobInfo(infoMessage);
+        handleJobRetry(message) {
+            Object.values(message).forEach((retryMessage) => {
+                // close all widgets tracking the previous job ID and restart them with the new widgets
+                const { job_id } = retryMessage;
+                const $currentRow = $(`#job_${job_id}`);
+                if ($currentRow && $currentRow[0].classList.contains('vertical_collapse--open')) {
+                    this.showHideChildRow($currentRow);
+                }
+                if (this.widgetsById[`${job_id}_LOG`]) {
+                    this.widgetsById[`${job_id}_LOG`].stop();
+                    this.widgetsById[`${job_id}_STATE`].stop();
+                }
             });
         }
 
-        _handleJobInfo(message) {
-            const jobId = message.job_id;
-            const jobState = this.jobManager.model.getItem(`exec.jobs.byId.${jobId}`);
+        /**
+         * parse and update the row with job info
+         * @param {object} message
+         */
+        handleJobInfo(message) {
             const appData = this.jobManager.model.getItem('app');
-            const rowIx = jobState && jobState.retry_parent ? jobState.retry_parent : jobId;
-
-            const jobDisplayData = generateJobDisplayData({
-                jobInfo: message,
-                fileTypeMapping: this.fileTypeMapping,
-                appData,
-                typesToFiles: this.config.typesToFiles,
+            Object.values(message).forEach((infoMessage) => {
```

Comment:
consolidate into a single function instead of calling `_handleJobInfo(infoMessage)`